Library/Homebrew/os/mac.rb

```diff
@@ -12,7 +12,19 @@ module Mac
     # This can be compared to numerics, strings, or symbols
     # using the standard Ruby Comparable methods.
     def version
-      @version ||= Version.new(MACOS_VERSION)
+      @version ||= Version.new(full_version.to_s[/10\.\d+/])
+    end
+
+    # This can be compared to numerics, strings, or symbols
+    # using the standard Ruby Comparable methods.
+    def full_version
+      @full_version ||= Version.new(
+        if RUBY_PLATFORM =~ /darwin/
```

Comment:
I think  it may be better to check OS type in `global.rb`. After all, it doesn't make sense to check OS type in `OS::Mac#full_version`.